python/cudf/cudf/core/column/timedelta.py

```diff
@@ -0,0 +1,402 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+import datetime as dt
+from numbers import Number
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+import cudf
+from cudf import _lib as libcudf
+from cudf._lib.nvtx import annotate
+from cudf._lib.scalar import Scalar, as_scalar
+from cudf.core.buffer import Buffer
+from cudf.core.column import column
+from cudf.core.column.datetime import _numpy_to_pandas_conversion
+from cudf.utils.dtypes import is_scalar, np_to_pa_dtype
+
+
+class TimeDeltaColumn(column.ColumnBase):
+    def __init__(
+        self, data, dtype, mask=None, size=None, offset=0, null_count=None
+    ):
+        """
+        Parameters
+        ----------
+        data : Buffer
+            The Timedelta values
+        dtype : np.dtype
+            The data type
+        mask : Buffer; optional
+            The validity mask
+        size : int, optional
+            Size of memory allocation.
+        offset : int
+            Data offset
+        null_count : int, optional
+            The number of null values.
+            If None, it is calculated automatically.
+        """
+        dtype = np.dtype(dtype)
+        if data.size % dtype.itemsize:
+            raise ValueError("Buffer size must be divisible by element size")
+        if size is None:
+            size = data.size // dtype.itemsize
+            size = size - offset
+        super().__init__(
+            data,
+            size=size,
+            dtype=dtype,
+            mask=mask,
+            offset=offset,
+            null_count=null_count,
+        )
+        assert self.dtype.type is np.timedelta64
+        self._time_unit, _ = np.datetime_data(self.dtype)
+
+    def __contains__(self, item):
+        # Handles improper item types
+        try:
+            item = np.timedelta64(item, self._time_unit)
+        except Exception:
+            return False
+        return item.astype("int_") in self.as_numerical
+
+    def to_pandas(self, index=None, nullable_pd_dtype=False):
+        return pd.Series(
+            self.to_array(fillna="pandas").astype(self.dtype), index=index
+        )
```

Comment:
This `True` value propagates down to this function: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/series.py#L2009

This will probably force users to pass `sr.to_pandas(nullable_pd_dtype=False)`, it is non-functional but will involve user to force send this param as False because we are internally defaulting to `True`

Reply:
We could dispatch that value based on the dtype, i.e. default to `False` for datetime / float types, `True` for int / bool types

Reply:
Made this change 👍

Reply:
No need to worry about this too much, especially since there isn't nullable pandas timestamps afaik. I'm removing the kwarg from series and dataframe in my next PR, so anything that we need to write here to handle it won't be in the code for long.